### ISSUE TITLE
"Manage blog" is unavailable

### ISSUE THREAD

**Issue by rusangl** [scubajeff/pico_lespas#2]
Les Pas application with blogging feature looks very interesting.
But I followed all 5 steps from the instruction on my Nextcloud instance (25.0.6.1, NCP)  and "Manage Blog"  button is still grey and unclickable. 
And there is no related logs in Nextcloud.
Sites created with Pico CMS are reachable.
How to find what causes the problem?

**Comment by scubajeff** [comment]
Do you know how to read your web server's access log? If yes, please try grep `csrftoken` and `cms` from the log output, for example with Apache2:
```grep csrftoken /var/log/apache2/access.log```
```grep cms /var/log/apache2/access.log```

**Comment by rusangl** [comment]
I have just checked and my results with Apache2 are:

"GET /csrftoken HTTP/2.0" 404 1389 "-" "LesPas_v2.8.2"

"GET /index.php/apps/cms_pico/personal/websites HTTP/2.0" 412 1271 "-" "LesPas_v2.8.2"

**Comment by scubajeff** [comment]
Nextcloud doesn't provide API for user (except admin) to list apps installed on server. To find out if cms_pico is installed or not, Les Pas will try accessing pico's API `/index.php/apps/cms_pico/personal/websites`, but that API is guarded by CSRF token, that's why Les Pas will call Nextcloud's API endpoint `/csrftoken` to get that token first. From the log, we can see that the Nextcloud instance you are running refused call to `csrftoken` endpoint with `File not found` error 404.
Nextcloud's API is not very stable, however, I checked Nextlcoud's github repo just now, that endpoint is still listed in the core module's `routes.php` file, means it should still be available in the latest release. So I'm scratching my head now.

**Comment by scubajeff** [comment]
If you deem it appropriate, could you please tell me your instance's address, so that I can debug remotely. Send it to my email: scubajeffrey@protonmail.ch

**Comment by cybershaman** [comment]
Hello there!

Just want to chime in with a possible workaround as I'vehad this issue for some time now too.

It appears that the Nextcloud web client and desktop clients are polling the csrftoken differently and succeeding, i.e.:
`10.x.x.x - - [21/May/2023:11:28:10 +0200] "GET /index.php/csrftoken HTTP/2.0" 200`

whereas LesPas polls the following and fails:
`212.95.5.128 - - [21/May/2023:11:23:32 +0200] "GET /csrftoken HTTP/2.0" 404`

I've now successfully worked around this issue by including a rewrite rule on my apache reverse proxy like so:
`RewriteRule ^/csrftoken https://%{SERVER_NAME}/index.php/csrftoken [R=301,L]`

**Comment by scubajeff** [comment]
Cool! Thank you for letting me know this. A fix to this endpoint issue will be included in next release.

**Comment by scubajeff** [comment]
Fixed in release 2.8.4

**Comment by rusangl** [comment]
So, a new 2.8.4 version appeared on f-droid and I checked it.
"Manage blog" works now!

But I faced another problem with blogging. When I open a link, provided by Les Pas, I see "site not found".
In apache2 logs I found:

"PROPFIND /remote.php/dav/files/User1/lespas HTTP/2.0" 207 1635 "-" "LesPas_v2.8.4"
"MKCOL /remote.php/dav/files/User1/lespas/.__picoblog__ HTTP/2.0" 405 1343 "-" "LesPas_v2.8.4"

And __picoblog__ is empty. Is it a webdav problem? Can it be solved?

**Comment by scubajeff** [comment]
Sorry for the late reply! It's a webdav issue. I will check the code asap

**Comment by scubajeff** [comment]
>  "MKCOL /remote.php/dav/files/User1/lespas/.picoblog HTTP/2.0" 405 1343 "-" "LesPas_v2.8.4"

Are you sure the message said `.picoblog` instead of `.__picoblog__`? Because the later one is what Les Pas use to store all the files of your website pages. You can browse your server's Les Pas folder, and there should be a `.__picoblog__` sub-folder there, and within it you should see two sub-folder `assets` and `content`

**Comment by rusangl** [comment]
You are right about .__picoblog__, automat8c formatting made it bold in the post.
I checked this folder on server and it is empty

**Comment by scubajeff** [comment]
Try "Manage Blog" again in Les Pas, and monitor your http server's access log for call to `/index.php/apps/cms_pico/personal/websites`

**Comment by rusangl** [comment]
Seems to be OK: "GET /index.php/apps/cms_pico/personal/websites HTTP/2.0" 200 759 "-" "LesPas_v2.8.4"
